packages/agents/sequencer/src/lib/operations/execute/fastpath.ts

```diff
@@ -275,11 +275,25 @@ export const executeFastPathData = async (
     const totalBids = bidsRoundMap[roundIdInNum];
     const _combinedBidsForRound = getAllSubsets(totalBids, getMinimumBidsCountForRound(roundIdInNum)) as Bid[][];
     // shuffle the bids
-    const combinedBidsForRound = _combinedBidsForRound
+    const _shuffledCombinedBidsForRound = _combinedBidsForRound
       .map((value) => ({ value, sort: Math.random() }))
       .sort((a, b) => a.sort - b.sort)
       .map(({ value }) => value);
 
+    /**
+     * The number of combinations of `n` objects taken `r` at a time is determined by the following formula
+     *      C(n,r)=n!(n-r)!
+     * If n=10 and r=3, C(10,3) = 120 -- this means it should iterate 120 times regardless of calldata execution errors.
+     * It might result in timeout error in consumer and requeue the message infinitely.
+     *
+     * Limiting the iteration count can help prevent infinite loops
+     */
+
+    const combinedBidsForRound =
```

Comment:
1.It's difficult to guarantee 100% success. In 90% of cases, the success rate depends on the success of the first chosen bid. If the first bid is successful, it is then sent to the Gelato network. However, if the first bid fails, subsequent bids will also fail. Here, the number 3 is an experimental value.


2. Like the first one, if the initial three combinations fail, it will likely fail with subsequent combinations due to its inherent error. There might be edge cases where bids are missed, but we can ensure that this happens at a _**LOW**_ percentage.